src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -1241,18 +1220,8 @@ public static HttpClientHandler CreateHttpMessageHandler(X509Certificate2 certif
                 throw new NotSupportedException("HttpClient implementation for the current platform does not support request compression.");
             }
 
-            var serverCallBack = _serverCertificateCustomValidationCallback;
-            if (serverCallBack != null)
-            {
-                if (PlatformDetails.RunningOnMacOsx)
-                {
-                    throw new PlatformNotSupportedException("On Mac OSX, is it not possible to register to ServerCertificateCustomValidationCallback because of https://github.com/dotnet/corefx/issues/9728");
-                }
-                else
-                {
-                    httpMessageHandler.ServerCertificateCustomValidationCallback += OnServerCertificateCustomValidationCallback;
-                }
-            }
+            if (PlatformDetails.RunningOnMacOsx == false)
+                httpMessageHandler.ServerCertificateCustomValidationCallback += OnServerCertificateCustomValidationCallback;
```

Comment:
Needs error handling here. 
In particular, need to see if an exception is raised here. If so, need to raise it up if there are any registered callbacks and if there aren't, remember it and raise it when someone try to register.

Reply:
Can you check now?